### PR TITLE
[Onboarding] Select an NPO Page

### PR DESCRIPTION
**Pull request by torinnayak1**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Made the onboarding tab and added the onboardingflow with the select an npo picker

feel free to ignore the inline styling for the view, just couldnt see anything on the top of my screen without it, sorry yo

also can ignore the default screen being the onboarding page on the layout page, that was for testing without having to navigate.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

onboarding/page.tsx

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

styling +1 +1 +1 (omg thanks guys for all of the support)

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @rachaelch3n

### CODE REVIEW

**Review by stephenhungg — COMMENTED** [review]
Type type typeeee looks good (confirmed with AI)

**Review by jeannineruiz — APPROVED** [review]
LGTM

**Review by stephenhungg — APPROVED** [review]

File: src/app/_layout.tsx [hunk 1]
```diff
@@ -16,8 +17,12 @@ function IndexScreen() {
...
   return (
     <SafeAreaProvider>
-      <Stack.Navigator screenOptions={{ headerShown: false }}>
+      <Stack.Navigator
+        screenOptions={{ headerShown: false }}
+        initialRouteName="Onboarding"
+      >
         <Stack.Screen name="Index" component={IndexScreen} />
+        <Stack.Screen name="Onboarding" component={OnboardingFlow} />
       </Stack.Navigator>
```
Comment (rachaelch3n) [inline src/app/_layout.tsx]: this is kinda on me but im gonna copy and paste what kyle said on stephen's pr regarding a change needed for the stack navigation:

"i think you almost have the right idea for expo router stack navigation, but some tweaks need to be made:

1. add a _layout.tsx file to each of the new screen folders (/login and /signup **[/onboarding for you]**), and rename page.tsx to index.tsx
2. in the _layout.tsx file, you can past this as the content:
```
import { Stack } from 'expo-router';

function StackLayout() {
  return (
    <Stack>
      <Stack.Screen name="index" options={{ headerShown: false }} />
    </Stack>
  );
}

export default StackLayout;
```
this basically exports the index.tsx file as the screen itself, so we can go back to app/_layout.tsx and rename the Stacks to name="auth/login" and name="auth/signup" respectively.
3. this allows us to use router to route to auth/signup or auth/login instead **[/onboarding]** of adding the /page at the end, which is kinda bad practice

lmk if this doesn't make sense, if you need more examples take a look at [Girls Write Now's codebase](https://github.com/calblueprint/girls-write-now)"

**Review by rachaelch3n — CHANGES_REQUESTED** [review]

File: src/app/_layout.tsx [hunk 2]
```diff
@@ -16,8 +17,12 @@ function IndexScreen() {
 function StackLayout() {
   return (
     <SafeAreaProvider>
-      <Stack.Navigator screenOptions={{ headerShown: false }}>
+      <Stack.Navigator
+        screenOptions={{ headerShown: false }}
+        initialRouteName="Onboarding"
```
Comment (rachaelch3n) [inline src/app/_layout.tsx]: also can we get rid of the initial route? you should get a screen with site map that lets you route to your onboarding page when you run the app

**Review by rachaelch3n — CHANGES_REQUESTED** [review]

File: src/app/_layout.tsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { Text, View } from 'react-native';
 import { SafeAreaProvider } from 'react-native-safe-area-context';
```
Comment (rachaelch3n) [inline src/app/_layout.tsx]: sorry, i didn't catch this earlier but we will be using expo router! not react navigation. the nav flow you have right now is using react navigation. The layout should look something like this:

`function StackLayout() {
  return (
  <Stack>
      <Stack.Screen name="index" options={{ headerShown: false }} />
      <Stack.Screen name="Onboarding" options={{ headerShown: false }} />
    </Stack>
  `

**Review by rachaelch3n — CHANGES_REQUESTED** [review]
